src/redux/actions/projectActions.ts

```diff
@@ -24,11 +28,35 @@ export function loadProject(project: IProject): (dispatch: Dispatch) => Promise<
     };
 }
 
-export function saveProject(project: IProject): (dispatch: Dispatch) => Promise<IProject> {
-    return async (dispatch: Dispatch) => {
+export function saveProject(project: IProject):
+  (dispatch: Dispatch, getState: any) => Promise<IProject> {
+    return async (dispatch: Dispatch, getState: any) => {
         const projectService = new ProjectService();
-        project = await projectService.save(project);
-        dispatch(saveProjectAction(project));
+        const projectId = project["id"];
+        const projectName = project["name"];
+        const sourceConnection = project.sourceConnection.name;
+        const targetConnection = project.targetConnection.name;
+        const projectList = getState().recentProjects;
+        if (projectList && projectList.length > 0) {
+            const isUnique = (project.id === null) ||
```

Comment:
Won't this comparison of `project.id === null` bypass the duplicate check and allow a duplicate to be created?

Reply:
Yes you are right, I rewrote the logic in the projectservice